Scraping Google Play Reviews of an app and performing sentiment analysis using transformers(Sentiment). 

I have used the online fashion app MANGO!

In [ ]:
import pandas as pd
import numpy as np
from google_play_scraper import app, Sort, reviews_all;#Scraping data

In [ ]:
#adding the app id and other details
mango=reviews_all('com.dylvian.mango.activities',sleep_milliseconds=0,lang='en',sort=Sort.NEWEST)

In [ ]:
mango

[{'reviewId': 'b8c589b7-ff69-4b0c-9268-9b8b527c1629',
  'userName': 'yazan fateh',
  'userImage': 'https://play-lh.googleusercontent.com/a/AItbvmleTY3Gyd31gev40Kw6w67RmW5tetQ9HOwT7GOK=mo',
  'content': 'The best',
  'score': 5,
  'thumbsUpCount': 0,
  'reviewCreatedVersion': '22.17.00',
  'at': datetime.datetime(2022, 9, 3, 3, 46, 32),
  'replyContent': None,
  'repliedAt': None},
 {'reviewId': 'e909e1d2-d531-45fa-ac78-93bdf52770ee',
  'userName': 'Janet Latham',
  'userImage': 'https://play-lh.googleusercontent.com/a/AItbvmlC_8mRFEUbOuMpz2VkP_X0VAZVcYKXJwOdRAAn=mo',
  'content': 'Very easy and clear to use',
  'score': 5,
  'thumbsUpCount': 0,
  'reviewCreatedVersion': '22.17.00',
  'at': datetime.datetime(2022, 9, 2, 6, 23, 7),
  'replyContent': None,
  'repliedAt': None},
 {'reviewId': '23020814-2c23-40c8-b345-ef9433ba6a4e',
  'userName': 'HealthCoachingForLife',
  'userImage': 'https://play-lh.googleusercontent.com/a-/AFdZucohhoGmAIS0VFbUltO_fujKGXEpLAxHLZqNxXc-Wg',
  'content': 'A

In [ ]:
df=pd.json_normalize(mango)

In [ ]:
df.head(500)

reviewId                       userName  \
0    b8c589b7-ff69-4b0c-9268-9b8b527c1629                    yazan fateh   
1    e909e1d2-d531-45fa-ac78-93bdf52770ee                   Janet Latham   
2    23020814-2c23-40c8-b345-ef9433ba6a4e          HealthCoachingForLife   
3    4f169143-6d4f-4e0b-a4a8-6e002f64431a                     kim Morand   
4    5e64e7e9-348b-4c14-8135-1f277f75cb7e                     Ayush Sahu   
..                                    ...                            ...   
495  4dd6e301-62c0-4eab-beca-0d1bbcec34bd                   Manleen Sran   
496  9d9a120c-0425-4825-94ac-eae464d5c006                 Annette Matadi   
497  2f238111-0906-4845-8d20-9f6925e1ae1a              Gayatry Borgoyary   
498  d174ef5d-b346-4016-819a-4d9f3593dd4c  Maria Agnes Perpetua Lambuson   
499  a0309bb4-d734-4d42-b621-5316cb3f3c29                          maryb   

                                             userImage  \
0    https://play-lh.googleusercontent.com/a/AItbvm...   
1    https://play-lh.googleusercontent.com/a/AItbvm...   
2    https://play-lh.googleusercontent.com/a-/AFdZu...   
3    https://play-lh.googleusercontent.com/a/AItbvm...   
4    https://play-lh.googleusercontent.com/a-/AFdZu...   
..                                                 ...   
495  https://play-lh.googleusercontent.com/a-/AFdZu...   
496  https://play-lh.googleusercontent.com/a-/AFdZu...   
497  https://play-lh.googleusercontent.com/a/AItbvm...   
498  https://play-lh.googleusercontent.com/a-/AFdZu...   
499  https://play-lh.googleusercontent.com/a/AItbvm...   

                                               content  score  thumbsUpCount  \
0                                             The best      5              0   
1                           Very easy and clear to use      5              0   
2                                           Awesome!!!      5              0   
3    I had to remove certain items from my basket i...      1              0   
4                                                 Good      1              0   
..                                                 ...    ...            ...   
495                                               👌🏻👌🏻      5              0   
496  I gave it a 3 star because of the lack in size...      3              2   
497  Products are too good . But bhai ek panty ke 5...      2              0   
498  I spent shopping online on MANGO app (200k + )...      1              0   
499                          Easy and straight forward      5              0   

    reviewCreatedVersion                  at  \
0               22.17.00 2022-09-03 03:46:32   
1               22.17.00 2022-09-02 06:23:07   
2               22.17.00 2022-09-01 21:02:30   
3               22.17.00 2022-09-01 15:10:45   
4                   None 2022-09-01 00:53:44   
..                   ...                 ...   
495                 None 2021-05-24 18:55:39   
496             21.08.00 2021-05-19 23:36:30   
497             21.08.00 2021-05-18 19:21:41   
498                 None 2021-05-18 10:41:23   
499             21.07.00 2021-05-17 05:53:12   

                                          replyContent           repliedAt  
0                                                 None                 NaT  
1                                                 None                 NaT  
2                                                 None                 NaT  
3    Hi and thank you for your feedback. Kindly be ... 2022-09-02 10:44:30  
4                                                 None                 NaT  
..                                                 ...                 ...  
495                                               None                 NaT  
496  Hi and thank you for your feedback. We will ta... 2021-05-27 11:01:38  
497  Hi. We greatly value customer feedback, and th... 2021-05-27 10:31:11  
498  Hello Maria and sorry to hear this. We kindly ... 2021-05-27 09:38:14  
499                                   

In [ ]:
df['score'].mean()

3.506804572672836

In [ ]:
df['reviewCreatedVersion'].value_counts()

20.17.00    75
19.13.00    54
21.24.00    40
20.08.01    37
19.6.00     31
            ..
19.5.00      1
2.0          1
4.2.27       1
1.1          1
1.0          1
Name: reviewCreatedVersion, Length: 131, dtype: int64

In [ ]:
from transformers import pipeline
#using a pre-trained model from Huggging Face
sentiment_analysis=pipeline("sentiment-analysis",model="siebert/sentiment-roberta-large-english")

In [ ]:
print(sentiment_analysis("I Loved it!!"))

[{'label': 'POSITIVE', 'score': 0.9988237023353577}]


In [ ]:
df.dtypes

reviewId                        object
userName                        object
userImage                       object
content                         object
score                            int64
thumbsUpCount                    int64
reviewCreatedVersion            object
at                      datetime64[ns]
replyContent                    object
repliedAt               datetime64[ns]
dtype: object

In [ ]:
df['content']=df['content'].astype('str')

In [ ]:
#creating a new column to store the sentiment and the score
df['result']=df['content'].apply(lambda x: sentiment_analysis(x))

In [ ]:
df.head(500)

reviewId                       userName  \
0    b8c589b7-ff69-4b0c-9268-9b8b527c1629                    yazan fateh   
1    e909e1d2-d531-45fa-ac78-93bdf52770ee                   Janet Latham   
2    23020814-2c23-40c8-b345-ef9433ba6a4e          HealthCoachingForLife   
3    4f169143-6d4f-4e0b-a4a8-6e002f64431a                     kim Morand   
4    5e64e7e9-348b-4c14-8135-1f277f75cb7e                     Ayush Sahu   
..                                    ...                            ...   
495  4dd6e301-62c0-4eab-beca-0d1bbcec34bd                   Manleen Sran   
496  9d9a120c-0425-4825-94ac-eae464d5c006                 Annette Matadi   
497  2f238111-0906-4845-8d20-9f6925e1ae1a              Gayatry Borgoyary   
498  d174ef5d-b346-4016-819a-4d9f3593dd4c  Maria Agnes Perpetua Lambuson   
499  a0309bb4-d734-4d42-b621-5316cb3f3c29                          maryb   

                                             userImage  \
0    https://play-lh.googleusercontent.com/a/AItbvm...   
1    https://play-lh.googleusercontent.com/a/AItbvm...   
2    https://play-lh.googleusercontent.com/a-/AFdZu...   
3    https://play-lh.googleusercontent.com/a/AItbvm...   
4    https://play-lh.googleusercontent.com/a-/AFdZu...   
..                                                 ...   
495  https://play-lh.googleusercontent.com/a-/AFdZu...   
496  https://play-lh.googleusercontent.com/a-/AFdZu...   
497  https://play-lh.googleusercontent.com/a/AItbvm...   
498  https://play-lh.googleusercontent.com/a-/AFdZu...   
499  https://play-lh.googleusercontent.com/a/AItbvm...   

                                               content  score  thumbsUpCount  \
0                                             The best      5              0   
1                           Very easy and clear to use      5              0   
2                                           Awesome!!!      5              0   
3    I had to remove certain items from my basket i...      1              0   
4                                                 Good      1              0   
..                                                 ...    ...            ...   
495                                               👌🏻👌🏻      5              0   
496  I gave it a 3 star because of the lack in size...      3              2   
497  Products are too good . But bhai ek panty ke 5...      2              0   
498  I spent shopping online on MANGO app (200k + )...      1              0   
499                          Easy and straight forward      5              0   

    reviewCreatedVersion                  at  \
0               22.17.00 2022-09-03 03:46:32   
1               22.17.00 2022-09-02 06:23:07   
2               22.17.00 2022-09-01 21:02:30   
3               22.17.00 2022-09-01 15:10:45   
4                   None 2022-09-01 00:53:44   
..                   ...                 ...   
495                 None 2021-05-24 18:55:39   
496             21.08.00 2021-05-19 23:36:30   
497             21.08.00 2021-05-18 19:21:41   
498                 None 2021-05-18 10:41:23   
499             21.07.00 2021-05-17 05:53:12   

                                          replyContent           repliedAt  \
0                                                 None                 NaT   
1                                                 None                 NaT   
2                                                 None                 NaT   
3    Hi and thank you for your feedback. Kindly be ... 2022-09-02 10:44:30   
4                                                 None                 NaT   
..                                                 ...                 ...   
495                                               None                 NaT   
496  Hi and thank you for your feedback. We will ta... 2021-05-27 11:01:38   
497  Hi. We greatly value customer feedback, and th... 2021-05-27 10:31:11   
498  Hello Maria and sorry to hear this. We kindly ... 2021-05-27 09:38:14   
499                        

In [ ]:
y=sentiment_analysis("I Loved it!!")

In [ ]:
y[0]

{'label': 'POSITIVE', 'score': 0.9988237023353577}

In [ ]:
df['sentiment']=df['result'].apply(lambda x: (x[0]['label']))
df['score']=df['result'].apply(lambda x: (x[0]['score']))

In [ ]:
df.head(500)

reviewId                       userName  \
0    b8c589b7-ff69-4b0c-9268-9b8b527c1629                    yazan fateh   
1    e909e1d2-d531-45fa-ac78-93bdf52770ee                   Janet Latham   
2    23020814-2c23-40c8-b345-ef9433ba6a4e          HealthCoachingForLife   
3    4f169143-6d4f-4e0b-a4a8-6e002f64431a                     kim Morand   
4    5e64e7e9-348b-4c14-8135-1f277f75cb7e                     Ayush Sahu   
..                                    ...                            ...   
495  4dd6e301-62c0-4eab-beca-0d1bbcec34bd                   Manleen Sran   
496  9d9a120c-0425-4825-94ac-eae464d5c006                 Annette Matadi   
497  2f238111-0906-4845-8d20-9f6925e1ae1a              Gayatry Borgoyary   
498  d174ef5d-b346-4016-819a-4d9f3593dd4c  Maria Agnes Perpetua Lambuson   
499  a0309bb4-d734-4d42-b621-5316cb3f3c29                          maryb   

                                             userImage  \
0    https://play-lh.googleusercontent.com/a/AItbvm...   
1    https://play-lh.googleusercontent.com/a/AItbvm...   
2    https://play-lh.googleusercontent.com/a-/AFdZu...   
3    https://play-lh.googleusercontent.com/a/AItbvm...   
4    https://play-lh.googleusercontent.com/a-/AFdZu...   
..                                                 ...   
495  https://play-lh.googleusercontent.com/a-/AFdZu...   
496  https://play-lh.googleusercontent.com/a-/AFdZu...   
497  https://play-lh.googleusercontent.com/a/AItbvm...   
498  https://play-lh.googleusercontent.com/a-/AFdZu...   
499  https://play-lh.googleusercontent.com/a/AItbvm...   

                                               content     score  \
0                                             The best  0.998266   
1                           Very easy and clear to use  0.998859   
2                                           Awesome!!!  0.998680   
3    I had to remove certain items from my basket i...  0.998070   
4                                                 Good  0.998415   
..                                                 ...       ...   
495                                               👌🏻👌🏻  0.998462   
496  I gave it a 3 star because of the lack in size...  0.999505   
497  Products are too good . But bhai ek panty ke 5...  0.994119   
498  I spent shopping online on MANGO app (200k + )...  0.999499   
499                          Easy and straight forward  0.998665   

     thumbsUpCount reviewCreatedVersion                  at  \
0                0             22.17.00 2022-09-03 03:46:32   
1                0             22.17.00 2022-09-02 06:23:07   
2                0             22.17.00 2022-09-01 21:02:30   
3                0             22.17.00 2022-09-01 15:10:45   
4                0                 None 2022-09-01 00:53:44   
..             ...                  ...                 ...   
495              0                 None 2021-05-24 18:55:39   
496              2             21.08.00 2021-05-19 23:36:30   
497              0             21.08.00 2021-05-18 19:21:41   
498              0                 None 2021-05-18 10:41:23   
499              0             21.07.00 2021-05-17 05:53:12   

                                          replyContent           repliedAt  \
0                                                 None                 NaT   
1                                                 None                 NaT   
2                                                 None                 NaT   
3    Hi and thank you for your feedback. Kindly be ... 2022-09-02 10:44:30   
4                                                 None                 NaT   
..                                                 ...                 ...   
495                                               None                 NaT   
496  Hi and thank you for your feedback. We will ta... 2021-05-27 11:01:38   
497  Hi. We greatly value customer feedback, and th... 2021-05-27 10:31:11   
498  Hello Maria and sorry to hear this. We kindly ... 2021-05-27 09:

In [ ]:
df['score'].mean()

0.9972433129859866

In [ ]:
df['sentiment'].value_counts()

POSITIVE    1084
NEGATIVE     753
Name: sentiment, dtype: int64

In [ ]:
df['sentiment'].value_counts(normalize=True)

POSITIVE    0.590093
NEGATIVE    0.409907
Name: sentiment, dtype: float64

**Observation**: From the above results we can see that the MANGO online
fashion app has more positive reviews than negative reviews. The percentage of positive reviews is 59%, whereas the percentage of negative reviews is 40%.